volatility3/framework/plugins/windows/amcache.py

```diff
@@ -0,0 +1,646 @@
+    Because all values could potentially be paged out of memory or malformed, they are all
+    a union between their expected value and `interfaces.renderers.BaseAbsentValue`.
+    """
+
+    entry_type: str
+    path: NullableString = renderers.NotApplicableValue()
+    company: NullableString = renderers.NotApplicableValue()
+    last_modify_time: NullableDatetime = renderers.NotApplicableValue()
+    last_modify_time_2: NullableDatetime = renderers.NotApplicableValue()
+    install_time: NullableDatetime = renderers.NotApplicableValue()
+    compile_time: NullableDatetime = renderers.NotApplicableValue()
+    sha1_hash: NullableString = renderers.NotApplicableValue()
+    service: NullableString = renderers.NotApplicableValue()
+    product_name: NullableString = renderers.NotApplicableValue()
+
+
+def _entry_sort_key(entry_tuple: Tuple[NullableString, _AmcacheEntry]) -> str:
+    """Sorts entries by program ID. This is broken out as a function here
+    to ensure consistency in sorting between the `group_by` and `sorted` function
+    invocations.
+    """
+    program_id, _ = entry_tuple
+    key = program_id if isinstance(program_id, str) else ""
+    return key
+
+
+def _get_string_value(
+    values: Dict[str, reg_extensions.CM_KEY_VALUE], name: str
+) -> NullableString:
+    try:
+        value = values[name]
+    except KeyError:
+        return renderers.NotAvailableValue()
+
+    data = value.decode_data()
+    if not isinstance(data, bytes):
+        return renderers.UnparsableValue()
+
+    return data.decode("utf-16le", errors="replace").rstrip("\u0000")
+
+
+def _get_datetime_filetime_value(
+    values: Dict[str, reg_extensions.CM_KEY_VALUE], name: str
+) -> NullableDatetime:
+    try:
+        value = values[name]
+    except KeyError:
+        return renderers.NotAvailableValue()
+
+    data = value.decode_data()
+    if not isinstance(data, int):
+        return renderers.UnparsableValue()
+
+    return conversion.wintime_to_datetime(data)
+
+
+def _get_datetime_utc_epoch_value(
+    values: Dict[str, reg_extensions.CM_KEY_VALUE], name: str
+) -> NullableDatetime:
+    try:
+        value = values[name]
+    except KeyError:
+        return renderers.NotAvailableValue()
+
+    data = value.decode_data()
+    if not isinstance(data, (int, float)):
+        return renderers.UnparsableValue()
+
+    try:
+        return datetime.datetime.fromtimestamp(float(data), datetime.timezone.utc)
+    except (ValueError, OverflowError, OSError):
+        return renderers.UnparsableValue()
+
+
+def _get_datetime_str_value(
+    values: Dict[str, reg_extensions.CM_KEY_VALUE], name: str
+) -> NullableDatetime:
+    try:
+        value = values[name]
+    except KeyError:
+        return renderers.NotAvailableValue()
+
+    data = value.decode_data()
+    if not isinstance(data, int):
+        return renderers.UnparsableValue()
+
+    if isinstance(data, str):
```

Comment:
This seems... strange?  Does the registry record datetime values as... strings?!?!  Without a timezone?!?!?